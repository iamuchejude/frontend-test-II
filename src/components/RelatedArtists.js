import React, { Component } from 'react';
import PropType from 'prop-types';

class RelatedArtists extends Component {
  constructor(props) {
    super(props);
    this.state = {
      artists: null,
    };
  }

  componentDidMount() {
    const { token } = localStorage;

    fetch(`https://api.spotify.com/v1/artists/1vCWHaC5f2uS3yhpwWbIA6/related-artists`, {
      headers: {
        'Authorization': `Bearer ${token}`
      }
    }).then(res => res.json())
    .then(data => { this.setState({ artists: data.artists.splice(0, 10) }); console.log(data.artists.splice(0, 10)) })
    .catch(err => console.log(err));
  }

  render() {
    return (
      <div className="related_artists">
        <div className="header"> Related Artists </div>
        {
          !this.state.artists ? 'loading....' : 
          this.state.artists.map(artist => {
            const { name, images, id, followers, type } = artist;
            return (
              <div className="singleRelatedArtist" key={ id } >
                <div className="photo">
                  <img src={ images[0].url } alt={ name } />
                  </div>
                  <div className="profile">
                    <ul>
                      <li className="name"> { name } </li>
                      <li className="followers"><i className="fa fa-user" /> { followers.total } Followers </li>
                      <li className="type"><span>{ type }</span></li>
                    </ul>
                </div>
              </div>
            );
          })
        }
      </div>
    );
  }
}

export default RelatedArtists;
