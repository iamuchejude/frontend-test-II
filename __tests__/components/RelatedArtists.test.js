import React from 'react';
import { shallow } from 'enzyme';
import RelatedArtists from '../../src/components/RelatedArtists';

describe('App components', () => {
  it('should render app', () => {
    const wrapper  = shallow(<RelatedArtists />);

    expect(wrapper).toMatchSnapshot()
  })
});
