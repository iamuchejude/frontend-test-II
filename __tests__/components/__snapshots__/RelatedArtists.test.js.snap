// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components should render app 1`] = `
<div
  className="related_artists"
>
  <div
    className="header"
  >
     Related Artists 
  </div>
  loading....
</div>
`;
